lightly_studio_view/src/lib/hooks/useGlobalStorage.ts

```diff
@@ -153,19 +153,28 @@ export const useGlobalStorage = () => {
         },
 
         // Individual sample annotation crop selection methods
-        toggleSampleAnnotationCropSelection: (annotationId: string) => {
+        toggleSampleAnnotationCropSelection: (datasetId: string, annotationId: string) => {
             selectedSampleAnnotationCropIds.update((state) => {
-                if (state.has(annotationId)) {
-                    state.delete(annotationId);
+                const annotations = state[datasetId] ?? new Set();
+
+                if (annotations?.has(annotationId)) {
```

Comment:
annotations will always be defined
```suggestion
                if (annotations.has(annotationId)) {
```